mubench.pipeline/tasks/implementations/detect_providing_patterns.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from os.path import exists, join
+from typing import Optional, List
+from urllib.error import URLError
+
+from data.detector import Detector
+from data.detector_execution import DetectorExecution
+from data.misuse import Misuse
+from data.misuse_compile import MisuseCompile
+from data.project_version import ProjectVersion
+from data.version_compile import VersionCompile
+from tasks.configurations.detector_interface_configuration import key_findings_file, key_run_file, key_detector_mode, \
+    key_training_src_path, key_training_classpath, key_target_src_path, key_target_classpath, key_dependency_classpath
+from utils.web_util import download_file
+
+RUN_MODE_NAME = "detect_only"
+DETECTOR_MODE = 1
+
+
+class DetectProvidingPatternsTask:
```

Comment:
We can probably move most of the logic into a superclass. `findings` and the is the only thing we really need in the `DetectorExecution`, since it is used by the filters.